src/control/actor/host/notify.go

```diff
@@ -57,27 +53,41 @@ func (c *HostNotifyCmd) listenCloseF(net network.Network, addr ma.Multiaddr) {
 
 func (c *HostNotifyCmd) connectedF(net network.Network, conn network.Conn) {
 	logrus.Info("connection detected: ", conn.RemotePeer().String())
-
-	// request metadata as soon as we connect to a peer
-	peerData, err := PollPeerMetadata(conn.RemotePeer(), c.Base, c.PeerMetadataState, c.Store, c.PeerStore)
+	h, _ := c.Host()
+	// Request the Host Metadata
+	hInfo := ReqHostInfo(context.Background(), h, conn)
+	// Request the BeaconMetadata
+	bMetadata, err := ReqBeaconMetadata(context.Background(), h, conn.RemotePeer())
+	if err != nil {
+		log.Warn(err)
+	}
+	// request BeaconStatus metadata as we connect to a peer
+	bStatus, err := ReqBeaconStatus(context.Background(), h, conn.RemotePeer())
+	if err != nil {
+		log.Warn(err)
+	}
 	var peer metrics.Peer
-	if err == nil {
-		peer = fetchPeerExtraInfo(peerData)
-		_ = peer
-	} else {
-		log.Info("could not get metadata for peer: ", conn.RemotePeer().String(), " err: ", err)
-		// TODO: Add also IP taken from ENR of discovered peers
-		peer = metrics.Peer {
-			PeerId: conn.RemotePeer().String(),
-		}
+	// Read ENR of the Peer from the generated enode
+	n := c.Store.LatestENR(conn.RemotePeer())
+	if n.ID().String() == "" {
+		// TODO: If the peer wasn't discovered via dv5 "n" will be empty
+		// we could compose new ENR with the beacon metadata and the host info
+		log.Warn("Peer ENR not found")
 	}
+	hInfo.NodeID = n.ID().String()
+	// fetch all the info gathered from the peer into a new Peer struct
+	peer.FetchHostInfo(hInfo)
+	// TODO: Switch from Update ot Add in the future, when implementing metrics versions
+	peer.UpdateBeaconStatus(bStatus)
+	peer.UpdateBeaconMetadata(bMetadata)
+	// So far, just act like if we got new info, Update or Aggregate new info from a peer already on the Peerstore
 	c.PeerStore.AddPeer(peer)
 	c.PeerStore.AddConnectionEvent(conn.RemotePeer().String(), "Connection")
```

Comment:
small nitpick, but now that i see it. We can also directly call `peer.ConnectionEvent` before adding the peer to the `peerstore`. Cleaner imho:

```
peer.FetchHostInfo(hInfo)
peer.UpdateBeaconStatus(bStatus)
peer.UpdateBeaconMetadata(bMetadata)
peer.ConnectionEvent(x, y)
```